core/src/main/java/org/mule/runtime/core/internal/execution/MuleMessageProcessingManager.java

```diff
@@ -34,26 +38,34 @@
   @Inject
   private PolicyManager policyManager;
 
+  private Registry registry;
+
+  private Collection<MessageProcessPhase> registryMessageProcessPhases;
+
   @Override
   public void processMessage(MessageProcessTemplate messageProcessTemplate, MessageProcessContext messageProcessContext) {
     phaseExecutionEngine.process(messageProcessTemplate, messageProcessContext);
   }
 
-  @Override
+  @Inject
   public void setMuleContext(MuleContext context) {
     this.muleContext = context;
   }
 
+  @Inject
+  public void setRegistryMessageProcessPhases(Optional<Collection<MessageProcessPhase>> registryMessageProcessPhases) {
+    // This needs to be an Optional due to https://jira.spring.io/browse/SPR-15338
```

Comment:
Does this need a TODO to change it once the referenced issue is fixed?

Reply:
No, it won;t be fixed till Spring 5 at least. It's just that at first i tried without the `Optional` and it failed and i investigated, Just added the comment to discourage anyone to try to simplify this.